MovingBlocks/Terasology
# More work on Lighting Nodes

### Contains
- existing DirectionalLightsNode has been split in DeferredMainLightNode and ApplyDeferredLightingNode. Both nodes now almost fully take advantage of the new architecture. [1]
- DeferredMainLightNode work done in commits 2 and 3 (eea3f7c and dc3d9cb).
- ApplyDeferredLightingNode work done in commits 4 and 5 (c377647 and 62affd8).
- Added classes SetInputTexture (StateChange) and SetInputTextureTask. [2]
- removed renderLightComponent method from WorldRenderer and WorldRendererImpl.

### How to test
- start Terasology
- create or load a game
- verify rendering of the 3d scene continues to work as expected

### Footnotes
[1] _Almost_ fully because there are a few things that are not yet in StateChanges but should, for example the binding of default FBOs
[2] StateChange subclass SetInputTexture is a good start as it makes texture inputs used by a shader very explicit. However, it seems to me that without some sort of orchestrating manager textures are bound to and unbound from the GPU's texture units without any criteria other than "I need this texture available now". I might write an issue about this: ideally frequently used textures are permanently bound to a specific texture unit on the GPU, while infrequently used textures share the remaining pool of texture units.
